### DISCUSSION TITLE
How do I get the path to Graphics

### DISCUSSION THREAD

**Discussion by lykl** [pixijs/pixijs#9057] [Q&A] [answered]
I tried to use `currentPath`, but it returned null

**Comment by SuperSodaSea** [comment]
```js
const graphics = new PIXI.Graphics();
graphics.moveTo(10, 10).lineTo(10, 20).lineTo(20, 20);
console.log(graphics.currentPath);
```

![image](https://user-images.githubusercontent.com/8724868/211966213-d272495c-08e2-481f-8eb7-a354601ece24.png)

You need to draw something first, or `Graphics.currentPath` would be `null`.

**Reply by lykl** [reply]
I've drawn something, but it's still null.

Let me tell you about my scene

I create a shape with a mouse click and store the shape in a history stack

Draws the path of the mouse as the mouse moves

The drawing ends when the mouse is raised

When I try to get the path to the shape in the stack, it returns null

But when I print the log after the draw, it has a value, whereas the shape in the stack does not. The reference type is not overwritten directly, and the modification to its properties should be available, which confused me

**Comment by SuperSodaSea** [comment, marked as answer]
Oh sorry, it should be something like this:

```js
const graphics = new PIXI.Graphics();
graphics.moveTo(10, 10).lineTo(10, 20).lineTo(20, 20).closePath();
console.log(graphics.geometry.graphicsData[0].shape);
```

![image](https://user-images.githubusercontent.com/8724868/211970240-bb9bcdc9-4ace-46cd-acf7-ca28d8271a96.png)

**Reply by lykl** [reply]
Thank you very much, but I don't understand why `currentPath` is null.
A typescript error occurred at the same time:
```
graphics.geometry.graphicsData[0].shape.points

Attribute "points" does not exist on type "IShape".
There is no property "points" on the type "Rectangle".
```

**Reply by SuperSodaSea** [reply]
It seems that `Graphics.currentPath` is used to store the path that is currently being created. When `closePath()` or `endFill()` is called, it will be set to `null`.

---

For the TypeScript issue, you need to cast `GraphicsData.shape` to `Polygon`:

```ts
import type { Polygon } from `@pixi/core`;

(graphics.geometry.graphicsData[0].shape as Polygon).points
```

**Reply by lykl** [reply]
> closePath
That might be the reason, but I didn't call `closePath()` or `endFill()`
I read part of the source code and do clear currentPath on these two function calls,
The problem arises because the documentation does not explicitly state this

**Reply by SuperSodaSea** [reply]
Or when the `Graphics` is being rendered (in `_render()`), `finishPoly()` will be called and `currentPath` will also set to `null`.

**Reply by lykl** [reply]
> Or when the `Graphics` is being rendered (in `_render()`), `finishPoly()` will be called and `currentPath` will also set to `null`.

This seems to be the case, and I recommend either that `currentPath` be `protected` or that the considerations for using it be made clear in the documentation,or it is recommended to get the path using the method used here
